dartagnan/src/main/java/com/dat3m/dartagnan/expression/INonDet.java

```diff
@@ -9,22 +9,21 @@
 // TODO why is INonDet not a IConst?
 public class INonDet extends IExpr {
 
-    private final String id;
+    private final int id;
     private final boolean signed;
-    private final BigInteger min;
-    private final BigInteger max;
+    private BigInteger min;
+    private BigInteger max;
+    private String sourceName;
 
-    // Should only be accessed from Program
-    public INonDet(int id, IntegerType type, boolean signed, BigInteger min, BigInteger max) {
+    // Should only be accessed from ProgramBuilder
+    public INonDet(int id, IntegerType type, boolean signed) {
```

Comment:
Is this comment reasonable? What if some passes generate non-deterministic choices like, e.g., the compilation of Linux CAS?
I think it is appropriate to have the `Program` be able to generate new constants and the `ProgramBuilder` just invokes that functionality.

Reply:
I agree, it should be going to read `Program or ProgramBuilder`, but currently, this is correct as it is.  I plan to tinker with the life time of `Program` as in #442 in a future PR.  We should avoid two `INonDet` managers.

Reply:
We should definitely avoid two `INonDet` managers, but then why not have it only in the `Program`? All functionality on `ProgramBuilder` "dies" (becomes inaccessible) as soon as the initial program is constructed. So by default I would rather put such functionality into the `Program` if there is reason to access it later.
Linux CAS compilation does exactly that (well, it uses `BNonDet`, but still).

Anyway, the code can stay as it is, but I would have just changed/removed the comment (you don't need to, it's just a minor concern).